### ISSUE TITLE
DSpot plugin: use DSpot Maven interface instead of the CLI interface

### ISSUE THREAD

**Issue by jesus-gorronogoitia** [STAMP-project/stamp-ide#39]
#### Characteristics
- **Issue Type**: feature
- **Reproducibility**: [always, sometimes, random]
- **Severity**: feature
- **Tool/Service/Component**: DSpot plugin for Eclipse
- **Execution Environment**: All
- **Reporter**: Jesús Gorroñogoitia<jesus.gorronogoitia@atos.net>
#### Description
Current DSpot plugin for Eclipse invokes DSpot using its CLI. This forces the plugin to embed the DSpot library, and this is incompatible with the DSpot LGPL license.
Therefore, DSpot plugin for Eclipse should invoke DSpot using its Maven interface.
This has two benefits. 
1- The plugin does not ship the DSpot library
2- Using the latest version of DSpot only requires to modify its version in the generated pom.xml file.
In order to use DSpot with Maven interface, follows the following documentation:
[DSpot documentation](https://github.com/STAMP-project/dspot)

The way DSpot plugin should work now with DSpot is the following:
1- get Wizard configuration and generate a temporary dspot_pom_<configuration_name>.xml
2- run DSpot using Maven and that configuration.
3- Update Dspot configuration using either the generated pom or the wizard configuration

#### [Help on issue template](https://github.com/STAMP-project/eclipse-ide/blob/master/.github/ISSUE_DOC.md) 
Preview to follow the link or open file .github/ISSUE_DOC.md

**Comment by danglotb** [comment]
Hi @jesus-gorronogoitia 

You can run the maven plugin of DSpot without touching the pom of the targeted project.

If fact, all the CLI options are reused in the maven plugin parameters (you "just" need to add `-D` before, _e.g._ `--iteration 1` would give `-Diteration=1`).

So according to the issue, you just need to modify the java command by the call of the maven goal. Moreover, when you do not give the version of the plugin on the command line, it will retrieve the latest released on maven central.

To sum up, if I get it right, you can translate the following command line:
```sh
java -jar dspot.jar \
    --amplifiers MethodAdd:MethodRemove \
    --iteration 1 \
    --test eu.stamp_project.examples.dhell.HelloAppTest
```
by the following maven goal:
```sh
mvn eu.stamp-project:dspot-maven:amplify-unit-tests \
    -Damplifiers=MethodAdd,MethodRemove \
    -Diteration=1 \
    -Dtest=eu.stamp_project.examples.dhell.HelloAppTest
```

**Comment by jesus-gorronogoitia** [comment]
Bonjour @danglotb, merci beaupcoup for your advices, we will follow them.

**Comment by RicardoTej** [comment]
Perfect, I'm going to work on it
